app/services/group_export_service.rb

```diff
@@ -0,0 +1,68 @@
+class GroupExportService
+  RELATIONS = %w[
+    all_users
+    all_events
+    all_notifications
+    all_documents
+    all_reactions
+    memberships
+    membership_requests
+    discussions
+    polls
+    poll_options
+    poll_did_not_votes
+    poll_unsubscriptions
+    outcomes
+    stances
+    stance_choices
+    discussion_readers
+    comments
+  ]
+
+  JSON_PARAMS = { groups: {methods: [:type]},
+                  users:  {except: [:encrypted_password,
+                                    :reset_password_token,
+                                    :email_api_key,
+                                    :reset_password_token,
+                                    :unsubscribe_token] }}.with_indifferent_access.freeze
+
+  def self.export(groups, filename)
+    ids = Hash.new { |hash, key| hash[key] = [] }
+    File.open(filename, 'w') do |file|
+      groups.each do |group|
+        puts_record(group, file, ids)
+        RELATIONS.each do |relation|
+          puts "Exporting: #{relation}"
+          group.send(relation).find_each(batch_size: 20000) do |record|
+            puts_record(record, file, ids)
+          end
+        end
+      end
+    end
+  end
+
+  def self.export_filename_for(group)
+    "tmp/#{DateTime.now.strftime("%Y-%m-%d_%H-%M-%S")}_#{group.name.parameterize}.json"
```

Comment:
A note that the group.name.parameterize will fail for guest groups, which we should be able to export just fine.